src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -324,31 +324,41 @@ bool operator==(const ArrayBase &lhs, const ArrayBase &rhs) {
   return (std::memcmp(lhs.getBuffer(), rhs.getBuffer(),
                       lhs.getCount() * BasicType::getSize(lhs.getType())) == 0);
 }
+
+template<typename T>
+bool compare_array_0_and_non0s_helper(T ptr, const Byte *v, size_t size) {
```

Comment:
nit, leave as is, but next time please keep the compareArrayZerosAndNonZerosHelper_ naming convention